manifests/server/config.pp

```diff
@@ -1,11 +1,28 @@
 class kafka::server::config {
+   $upstart_file   = '/etc/init/kafka.conf'
 
-  file { '/etc/init/kafka.conf':
-    ensure => 'file',
-    owner  => 'root',
-    group  => 'root',
-    mode   => '0444',
-    content => template('kafka/kafka.conf.erb'),
+  if $::lsbdistcodename == 'trusty' {
+    file { $upstart_file:
+      ensure => 'file',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0444',
+      content => template('kafka/kafka.conf.erb'),
+    }
+  } else {
+    file { '/etc/init/pre-start.sh':
+      ensure => 'file',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0444',
```

Comment:
ah, i think you need an +x bit since this is a script.

Maybe something like `mode => '0544'` ?